teku/src/main/java/tech/pegasys/teku/cli/subcommand/debug/DebugToolsCommand.java

```diff
@@ -80,4 +86,53 @@ public int autocomplete(
       return 1;
     }
   }
+
+  @Command(
+      name = "generate-swagger-docs",
+      description = "Generate swagger-docs for rest APIs.",
+      mixinStandardHelpOptions = true,
+      showDefaultValues = true,
+      abbreviateSynopsis = true,
+      versionProvider = PicoCliVersionProvider.class,
+      synopsisHeading = "%n",
+      descriptionHeading = "%nDescription:%n%n",
+      optionListHeading = "%nOptions:%n",
+      footerHeading = "%n",
+      footer = "Teku is licensed under the Apache License 2.0")
+  public int generateSwaggerDocs(
+      @Option(
+              required = true,
+              names = {"--output", "-o"},
+              description = "Directory to write swagger docs to.")
+          final Path outputPath)
+      throws Exception {
+    if (!outputPath.toFile().mkdirs() && !outputPath.toFile().isDirectory()) {
+      throw new InvalidConfigurationException(
+          String.format(
+              "Destination path %s could not be created or is not a directory",
+              outputPath.toAbsolutePath()));
+    }
+    ValidatorRestApiConfig config =
+        ValidatorRestApiConfig.builder().restApiDocsEnabled(true).build();
+
+    final KeyManager keyManager = new NoOpKeyManager();
+    RestApi api = ValidatorRestApi.create(config, keyManager);
+
+    api.getRestApiDocs()
+        .ifPresentOrElse(
+            (docs) -> {
+              final Path validatorApiPath = outputPath.resolve("validator-api.json");
+              System.out.println("Writing validator-api to " + validatorApiPath.toAbsolutePath());
+              try (FileWriter fileWriter =
+                  new FileWriter(validatorApiPath.toFile(), StandardCharsets.UTF_8)) {
+                fileWriter.write(docs);
+              } catch (IOException e) {
+                System.err.println("Failed to write validator-api.json: " + e.getMessage());
```

Comment:
we should ensure we return a non-zero exit code on any failure (here and the `orElse` case if the api docs are unavailable), otherwise CI won't know it failed.